File: src/python/leetcode/build_array_from_permutation/buildArrayFromPermutation.py
#
# @lc app=leetcode id=1920 lang=python3
#
# [1920] Build Array from Permutation
#
from typing import List


# @lc code=start
class Solution:
    def buildArray(self, nums: List[int]) -> List[int]:
        answer: List[int] = []

        for i in range(0, len(nums)):
            answer.append(nums[nums[i]])

        return answer


# @lc code=end
s: Solution = Solution()
answer: List[int] = s.buildArray([0, 2, 1, 5, 3, 4])
print(answer)

File: src/typescript/leetcode/build_array_from_permutation/buildArrayFromPermutation.ts
/*
 * @lc app=leetcode id=1920 lang=typescript
 *
 * [1920] Build Array from Permutation
 */
// @lc code=start
function buildArray(nums: number[]): number[] {
  let built: number[] = [];

  for (let i = 0; i < nums.length; i++) built.push(nums[nums[i]]);

  return built;
}
// @lc code=end
console.log(buildArray([0, 2, 1, 5, 3, 4]));
